From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 3 Sep 2010 16:45:04 -0400
Subject: [net] bnx2x: fix wrong return from bnx2x_trylock_hw_lock
Message-id: <20100903164504.25415.61795.stgit@brian.englab.brq.redhat.com>
Patchwork-id: 28120
O-Subject: [RHEL5.6 BZ572012 PATCH 46/46] bnx2x: fix wrong return from
	bnx2x_trylock_hw_lock
Bugzilla: 572012
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Stanislaw Gruszka <sgruszka@redhat.com>

From: Eric Dumazet <eric.dumazet@gmail.com>

bnx2x_trylock_hw_lock() returns a bool :

true if succeeded to acquire the lock.
false in case of error.

-EINVAL is not an acceptable value, since its promoted to true.

Signed-off-by: Eric Dumazet <eric.dumazet@gmail.com>
Acked-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 0fdf4d096102464e290d5ccaee8ad0cee7bd9e8a)

diff --git a/drivers/net/bnx2x/bnx2x_main.c b/drivers/net/bnx2x/bnx2x_main.c
index 391611c..edff0d2 100644
--- a/drivers/net/bnx2x/bnx2x_main.c
+++ b/drivers/net/bnx2x/bnx2x_main.c
@@ -774,7 +774,7 @@ static bool bnx2x_trylock_hw_lock(struct bnx2x *bp, u32 resource)
 		DP(NETIF_MSG_HW,
 		   "resource(0x%x) > HW_LOCK_MAX_RESOURCE_VALUE(0x%x)\n",
 		   resource, HW_LOCK_MAX_RESOURCE_VALUE);
-		return -EINVAL;
+		return false;
 	}
 
 	if (func <= 5)
